### PR TITLE
Add support for optional serializer generation.

### PR DESCRIPTION
**Pull request by tkieft**
This will allow the user to enable or disable the generation of serializer code on a per-class or global basis, saving on code size and method count.

### CODE REVIEW

File: processor/build.gradle [hunk 1]
```diff
@@ -8,13 +8,14 @@ repositories {
 }
 
 dependencies {
-  compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.1+'
```
Comment (ttung) [inline processor/build.gradle]: any reason for this change?

File: processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java [hunk 2]
```diff
@@ -2,11 +2,7 @@
 
 package com.instagram.common.json.annotation.processor;
 
-import javax.annotation.processing.AbstractProcessor;
-import javax.annotation.processing.Filer;
-import javax.annotation.processing.Messager;
-import javax.annotation.processing.ProcessingEnvironment;
-import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.*;
```
Comment (ttung) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java]: any reason for this change?

File: processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java [hunk 3]
```diff
@@ -41,12 +37,16 @@
  * This annotation processor is run at compile time to find classes annotated with {@link JsonType}.
  * Deserializers are generated for such classes.
  */
+@SupportedOptions({"generateSerializers"})
```
Comment (ttung) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java]: mildly concerned that `generateSerializers` is a very generic name, but probably not a big deal.

File: processor/src/main/java/com/instagram/common/json/annotation/processor/JsonParserClassData.java [hunk 4]
```diff
@@ -184,39 +191,41 @@ public void emitJava(JavaWriter writer) throws IOException {
             .emitEmptyLine();
       }
 
-      writer
```
Comment (ttung) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonParserClassData.java]: ugh, i miss phabricator.

File: processor/testuut/parent/build.gradle [hunk 5]
```diff
@@ -4,6 +4,7 @@ compileJava {
   options.compilerArgs += [
                        '-processor',
                        'com.instagram.common.json.annotation.processor.JsonAnnotationProcessor'
+
```
Comment (ttung) [inline processor/testuut/parent/build.gradle]: ?

**Comment by ttung** [comment]
You could save just as many methods if you made the String <-> Object serializer/deserializer optional. :)

File: processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java (hunk 2 above)
Comment (tkieft) [inline processor/src/main/java/com/instagram/common/json/annotation/processor/JsonAnnotationProcessor.java]: IntelliJ did it, I can revert

File: processor/build.gradle (hunk 1 above)
Comment (tkieft) [inline processor/build.gradle]: Either IntelliJ or Gradle did this...

**Comment by tkieft** [comment]
String <-> Object do you think that's actually a good idea? The reason I did this is that we rarely use serializers, and this also saves us from writing serializeCodeFormatters.

**Comment by tkieft** [comment]
I guess we mostly use the parseFromJson(jp) version, but for serializers I imagine the String version is more widely used.

**Comment by ttung** [comment]
We mostly just use `parseFromJson(jp)`, and we mostly just use `serializeToJson(JsonGenerator,...`

Only the top-level objects would ever use the string versions.

**Comment by tkieft** [comment]
@ttung good point. I suppose that could be a separate feature request :)

**Comment by tkieft** [comment]
ping?

**Comment by ttung** [comment]
Sorry.  Github is not the greatest at notifying when a PR has been updated.  As in it doesn't.

**Pull request merged**